### PR TITLE
New Rule: Open Redirect: Google domain with /url path and suspicious indicators

### PR DESCRIPTION
**Pull request by morriscode**

### CODE REVIEW

**Comment by morriscode** [comment]
No firings yet.

File: detection-rules/link_google_open_redirect_with_suspicious_indicators.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: "Open Redirect: Google domain with /url path and suspicious indicators"
+description: |
+  This rule examines messages containing image attachments that utilize Google's open redirect (Google.com/url..).
```
Comment (jkamdjou) [inline detection-rules/link_google_open_redirect_with_suspicious_indicators.yml:3]:
```suggestion
  This rule examines messages containing image attachments that utilize Google's open redirect (google[.]com/url...). 
```

File: detection-rules/link_google_open_redirect_with_suspicious_indicators.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          and regex.contains(.href_url.query_params, "hl=.{2}&q=http(s)?://")
+  )
+  and 2 of (
+    // Not a google logo
+    any(attachments,
+        .file_type in ('png', 'jpg', 'bmp')
+        and (
+          any(ml.logo_detect(.).brands, not strings.starts_with(.name, "Google"))
+          or any(ml.logo_detect(beta.message_screenshot()).brands,
+                 not strings.starts_with(.name, "Google")
```
Comment (jkamdjou) [inline detection-rules/link_google_open_redirect_with_suspicious_indicators.yml:32]: does this return true or false if no brands are detected? if unsure, could make the explicit in the logic

File: detection-rules/link_google_open_redirect_with_suspicious_indicators.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          )
+        )
+    ),
+    // Body analysis - NLU - Credential theft
+    (
+      any(ml.nlu_classifier(body.html.inner_text).intents,
```
Comment (jkamdjou) [inline detection-rules/link_google_open_redirect_with_suspicious_indicators.yml:38]:
```suggestion
      any(ml.nlu_classifier(body.current_thread.text).intents,
```

File: detection-rules/link_google_open_redirect_with_suspicious_indicators.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      any(attachments,
+          .file_type in ('png', 'jpg', 'bmp')
+          and any(file.explode(.),
+                  any(ml.nlu_classifier(.scan.ocr.raw).intents, .name == "cred_theft")
+          )
+      )
+    ),
+    // Content analysis - Body - Urgency
+    (any(ml.nlu_classifier(body.html.inner_text).entities, .name == "urgency")
+    ),
```
Comment (jkamdjou) [inline detection-rules/link_google_open_redirect_with_suspicious_indicators.yml:53]:
```suggestion
    (
        any(ml.nlu_classifier(body.current_thread.text).entities, .name == "urgency")
    ),
```

File: detection-rules/link_google_open_redirect_with_suspicious_indicators.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  This rule examines messages containing image attachments that utilize Google's open redirect (Google.com/url..). 
+  To enhance accuracy and minimize false positives, the rule conducts additional assessments for suspicious indicators, as indicated in the comments.
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  // All attachments are images
+  and length(attachments) > 0
+  and all(attachments, .file_type in ('png', 'jpg', 'bmp'))
+  and sender.email.domain.root_domain not in $org_domains
```
Comment (jkamdjou) [inline detection-rules/link_google_open_redirect_with_suspicious_indicators.yml:12]: do we need to exclude legit google senders?

**Review by jkamdjou — APPROVED** [review]
overall LGTM, a few suggestions and comments, will approve and defer to you on what to address then merge

File: detection-rules/link_google_open_redirect_with_suspicious_indicators.yml (hunk 2 above)
Comment (morriscode) [reply detection-rules/link_google_open_redirect_with_suspicious_indicators.yml:32]: It returns false.

File: detection-rules/link_google_open_redirect_with_suspicious_indicators.yml (hunk 5 above)
Comment (morriscode) [reply detection-rules/link_google_open_redirect_with_suspicious_indicators.yml:12]: It hasn't been an issue in the past, this [rule](https://github.com/sublime-security/sublime-rules/blob/8737eb18cce587e9442ed7d8b714af874ca4670d/detection-rules/link_google_amp_suspicious_indicators.yml) is very similar

**Pull request merged**